src/WebUI/Components/KubeSummary.tsx

```diff
@@ -104,6 +105,28 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
         return null;
     }
 
+    private _getReplicaSetPodDetails(): JSX.Element | null {
+        const selectedItem = this.state.selectedItem;
+        if (selectedItem) {
+            const selectedItemDeploymentUId: string = selectedItem.deploymentId.toLowerCase();
+            const selectedItemReplicaSetUId: string = selectedItem.replicaSetId.toLowerCase();
+            const replicas = this.state.replicaSetList;
+            const filteredReplicas = (replicas && replicas.items || []).filter(replica => {
+                return Utils.isOwnerMatched(replica.metadata, selectedItemDeploymentUId) && replica.metadata.uid.toLowerCase() === selectedItemReplicaSetUId;
+            });
+
+            if (filteredReplicas) {
+                return (<WorkloadPodsView
+                    metaData={filteredReplicas[0].metadata}
+                    template={filteredReplicas[0].spec && filteredReplicas[0].spec.template}
+                    kind={"ReplicaSet"}

```

Comment:
when we select an item we can have this rather hard coding

Reply:
I have added 'kind' to the types IServiceItem and IDeploymentReplicaSetItem